Lib/functools.py

```
@@ -882,6 +882,9 @@ def __init__(self, func):
         self.func = func
         self.attrname = None
         self.__doc__ = func.__doc__
+        func_isabstractmethod = getattr(func, '__isabstractmethod__')
```

Used like that, getattr() raises an exception if the attribute doesn't exist.

You can use:

```
try: self.__isabstractmethod__ = func.__isabstractmethod__
except AttributeError: pass
```

---

Ah, thank you. I opened the implementation in my IDE and it said `default=None` so I was mislead. Pushing the version you're suggesting